code/modules/shuttles/docking_beacon.dm

```diff
@@ -145,19 +158,70 @@
 		D.ui_interact(user)
 		return TOPIC_HANDLED
 
+	if(href_list["toggle_construction"])
+		construction_mode = !construction_mode
+		errors.Cut()
+		return TOPIC_REFRESH
+
+	if(href_list["change_color"])
+		var/new_color = input(user, "Choose a color.", "\the [src]", ship_color) as color|null
+		if(!CanInteract(usr,state))
+			return TOPIC_NOACTION
+		if(new_color && new_color != ship_color)
+			ship_color = new_color
+			to_chat(user, SPAN_NOTICE("You set \the [src] to create a ship with <font color='[ship_color]'>this color</font>"))
+			return TOPIC_HANDLED
+
+	if(href_list["change_ship_name"])
+		var/new_ship_name = sanitize(input(user, "Enter a new name for the ship:", "Change ship name.") as null|text)
+		if(!CanInteract(usr,state))
+			return TOPIC_NOACTION
+		if(!new_ship_name)
+			return TOPIC_HANDLED
+		if(length(new_ship_name) > MAX_NAME_LENGTH)
+			to_chat(user, SPAN_WARNING("That name is too long!"))
+			return TOPIC_HANDLED
+		ship_name = new_ship_name
+		return TOPIC_REFRESH
+
+	if(href_list["check_validity"])
+		if(!construction_mode)
+			return TOPIC_HANDLED
+		check_ship_validity(get_areas())
+		return TOPIC_REFRESH
+	
+	if(href_list["finalize"])
+		if(!construction_mode)
+			return TOPIC_HANDLED
+		var/confirm = alert(user, "This will permanently finalize the ship, are you sure?", "Ship finalization", "Yes", "No")
```

Comment:
Standard pattern for these is to default to `No` so people don't accidentally confirm.

Reply:
Might be fixed now? Not exactly sure what you mean since there is no default argument for alert

Reply:
When the alert pops up, the leftmost value is highlighted, and hitting space or enter will select it. I am requesting that you change `"Yes", "No"` to `"No", "Yes"` to avoid people accidentally confirming.